src/main/java/com/redhat/rcm/offliner/model/DownloadResult.java

```diff
@@ -26,26 +26,39 @@
 
     private Exception error;
 
+    private boolean avoided;
+
+    public DownloadResult( String originUrl, String path, Exception error, boolean avoided )
+    {
+        this.originUrl = originUrl;
+        this.path = path;
+        this.error = error;
+        this.avoided = avoided;
+    }
+
     public static DownloadResult success( String originUrl, String path )
     {
-        return new DownloadResult( originUrl, path, null );
+        return new DownloadResult( originUrl, path, null, false );
     }
 
     public static DownloadResult error( String path, Exception error )
     {
-        return new DownloadResult( null, path, error );
+        return new DownloadResult( null, path, error, false );
     }
 
-    public DownloadResult( String originUrl, String path, Exception error )
+    public static DownloadResult avoid( String path, boolean avoided )
     {
-        this.originUrl = originUrl;
-        this.path = path;
-        this.error = error;
+        return new DownloadResult( null, path, null, avoided );
     }
 
     public boolean isSuccess()
     {
-        return error == null;
+        return error == null && avoided == false;
+    }
+
+    public boolean isAvoided()
+    {
+        return avoided == true;
     }
```

Comment:
It's stylistic, but you don't really need to do a comparison...you could just return the value. Likewise, the comparison on 'avoided' in isSuccess() could just be `&& !avoided`.